src/server/debug.rs

```diff
@@ -326,6 +328,52 @@ impl Debugger {
         Ok(res)
     }
 
+    pub fn remove_failed_stores(&self, store_ids: Vec<u64>) -> Result<()> {
+        let wb = WriteBatch::new();
+        let handle = box_try!(get_cf_handle(self.engines.kv_engine.as_ref(), CF_RAFT));
+        let store_ids = HashSet::<u64>::from_iter(store_ids);
+        box_try!(self.engines.kv_engine.scan_cf(
+            CF_RAFT,
+            keys::REGION_META_MIN_KEY,
+            keys::REGION_META_MAX_KEY,
+            false,
+            &mut |key, value| {
+                let (_, suffix_type) = box_try!(keys::decode_region_meta_key(key));
+                if suffix_type != keys::REGION_STATE_SUFFIX {
+                    return Ok(true);
+                }
+
+                let mut region_state = RegionLocalState::new();
+                box_try!(region_state.merge_from_bytes(value));
+                if region_state.get_state() == PeerState::Tombstone {
+                    return Ok(true);
+                }
+
+                let mut new_peers = region_state.get_region().get_peers().to_owned();
+                new_peers.retain(|peer| !store_ids.contains(&peer.get_store_id()));
+                let new_peers_len = new_peers.len();
+                let old_peers_len = region_state.get_region().get_peers().len();
+
+                if new_peers_len < quorum(old_peers_len) {
```

Comment:
why use quorum here?

Reply:
Suppose stores `3, 4, 5` are failed, and we have a region in stores `1, 2, 3`, so the region still has avaliable replicas on store `1` and store `2`. So we don't need to modify any metadata for this region.